### PR TITLE
Removes Hulk

### PR DESCRIPTION
**Pull request by CoffeeDragon16**
## About The Pull Request

This totally removes Hulk from the game. Given that it's part of the wizard's mutate spell, this replaces the hulk in it with X-ray vision and TK.

## Why It's Good For The Game

Hulk is uninteresting and unfun. It gives you the ability to ignore several mechanics, and is most often used to grief. It contributes to Genetics' powergaming problem of getting permanent massive bonuses for little cost. Fighting hulk, and using it, is uninteresting. As hulk gives total effective immunity to stuns and stamina, you often have to escalate directly to lethals to deal with hulk. You're unreasonably strong against people, immune to non-lethal deescalation, and able to easily destroy objects. This often ends up with people taking hulk and becoming immune to sec, which trivializes them and ends in grief.

## Changelog
:cl:
del: Hulk is now totally removed from the game.
tweak: The Wizard's mutate spell now gives X-ray and TK instead of Hulk.
/:cl:

### CODE REVIEW

**Comment by ShibaInuLord** [comment]
> Given that it's part of the wizard's mutate spell, this replaces the hulk in it with X-ray vision and TK.
this is dumb

**Comment by Gogo9001** [comment]
Ok I understand removing it from genetics, but is there a way you could just remove it from genetics consoles instead? It works fine for the wiz spell since it's limited and keeping it in for admin memes is also quite nice.

File: code/game/objects/items/devices/scanners.dm [hunk 1]
```diff
@@ -355,9 +354,9 @@ GENE SCANNER
 			|| S.mutantappendix != initial(S.mutantappendix) \
 			|| S.flying_species != initial(S.flying_species)
 
-		render_list += "<span class='info ml-1'>Species: [S.name][mutant ? "-derived mutant" : ""]</span>\n"
-		render_list += "<span class='info ml-1'>Core temperature: [round(H.coretemperature-T0C,0.1)] &deg;C ([round(H.coretemperature*1.8-459.67,0.1)] &deg;F)</span>\n"
-	render_list += "<span class='info ml-1'>Body temperature: [round(M.bodytemperature-T0C,0.1)] &deg;C ([round(M.bodytemperature*1.8-459.67,0.1)] &deg;F)</span>\n"
+		render_list += "<span class='info m1 -1'>Species: [S.name]</span>\n"
+		render_list += "<span class='info m1 -1'>Core temperature: [round(H.coretemperature-T0C,0.1)] &deg;C ([round(H.coretemperature*1.8-459.67,0.1)] &deg;F)</span\n"
+	render_list += "<span class='info m1 -1'>Body temperature: [round(M.bodytemperature-T0C,0.1)] &deg;C ([round(M.bodytemperature*1.8-459.67,0.1)] &deg;F)</span\n"
```
Comment (Jared-Fogle) [inline code/game/objects/items/devices/scanners.dm:359]: What?

Comment (Jared-Fogle) [reply code/game/objects/items/devices/scanners.dm:359]: Why'd you change this to `m1 -1`? You're going to break the styling.

File: code/game/objects/items/implants/implantchair.dm [hunk 2]
```diff
@@ -171,7 +171,7 @@
 		return FALSE
 	H.set_species(/datum/species/human, 1)//lizards go home
 	purrbation_remove(H)//remove cats
-	H.dna.remove_all_mutations()//hulks out
+	H.dna.remove_all_mutations()//powergamers out
```
Comment (Jared-Fogle) [inline code/game/objects/items/implants/implantchair.dm:174]: Just remove the comment, neither this nor the old one helps make the code any more readable. Comments should explain why, not what.

**Comment by wesoda25** [comment]
Moths can’t get hulk right?

File: code/modules/antagonists/blob/structures/_blob.dm [hunk 3]
```diff
@@ -212,23 +212,6 @@
 	if(overmind)
 		overmind.blobstrain.extinguish_reaction(src)
 
-/obj/structure/blob/hulk_damage()
-	return 15
-
-/obj/structure/blob/attackby(obj/item/I, mob/user, params)
```
Comment (Jared-Fogle) [inline code/modules/antagonists/blob/structures/_blob.dm:218]: ummmm

File: code/modules/mob/living/carbon/alien/humanoid/caste/hunter.dm [hunk 4]
```diff
@@ -76,7 +76,6 @@
 			if(!blocked)
 				L.visible_message("<span class='danger'>[src] pounces on [L]!</span>", "<span class='userdanger'>[src] pounces on you!</span>")
 				L.Paralyze(100)
-				sleep(2)//Runtime prevention (infinite bump() calls on hulks)
```
Comment (Jared-Fogle) [inline code/modules/mob/living/carbon/alien/humanoid/caste/hunter.dm:79]: Did you test that this code is no longer necessary for preventing runtimes?

**Comment by CoffeeDragon16** [comment]
>this is dumb

If you have a better idea, let me know. I'm unsure of other strong mutations I could include to replace it.

>Moths can’t get hulk right?

Yeah, human only.

**Comment by Sparkezel** [comment]
Hulk should be traitor gear

**Comment by Makarov22** [comment]
"Hulk is uninteresting and unfun. It gives you the ability to ignore several mechanics, and is most often used to grief."
Let me guess, you got mad because someone killed you as a hulk in some previous round lol.

**Comment by Pugie** [comment]
stun immunity at the price of only using melee is probably the most fair trade of I've ever heard of ever and isn't op. If it becomes a problem just use lethals/bolas lol you need to stop being unrobust.

**Comment by Tlaltecuhtli** [comment]
>immune to non-lethal deescalation,

where is the problem?

**Comment by PolishUser** [comment]
Just shoot hulks with lethals, problem solved.

File: code/game/objects/items/implants/implantchair.dm (hunk 2 above)
Comment (CoffeeDragon16) [reply code/game/objects/items/implants/implantchair.dm]: Removed it.

**Comment by Naloacisamazing** [comment]
Hulks in no way are immune to sec. Bolas slow them so their only way to do damage is to slowly take off the bola while they are being shot at. The arm breaking of current hulks mean that the constant wall and item breaks is almost a non issue since most dont wanna break their arms just to get into a room

File: code/modules/antagonists/blob/structures/_blob.dm (hunk 3 above)
Comment (CoffeeDragon16) [reply code/modules/antagonists/blob/structures/_blob.dm:218]: Ah, accident. Undid that.

File: code/game/objects/items/devices/scanners.dm (hunk 1 above)
Comment (CoffeeDragon16) [reply code/game/objects/items/devices/scanners.dm]: Ah, typo. Should be fixed now.

**Comment by usnpeepoo** [comment]
adding onto what others said ^ hulk is pretty balanced considering you're restricted from using ranged weapons, just at the cost of no longer fearing the baton/flash. you still get slowed by bolas/dragnets/beartraps/disablers (you just dont go into stamcrit). You also are completely fucked by any competent MD/para with a mutadone syring egun completely removing the entire mutation+any others (which i've done numerous times to tiding little fucks who think hulk=immune to everything). This PR doesn't make any sense.

File: code/_onclick/other_mobs.dm [hunk 5]
```diff
@@ -24,9 +24,6 @@
 	var/obj/item/clothing/gloves/G = gloves // not typecast specifically enough in defines
 	if(proximity && istype(G) && G.Touch(A,1))
 		return
-	//This signal is needed to prevent gloves of the north star + hulk.
-	if(SEND_SIGNAL(src, COMSIG_HUMAN_EARLY_UNARMED_ATTACK, A, proximity) & COMPONENT_CANCEL_ATTACK_CHAIN)
```
Comment (Jared-Fogle) [inline code/_onclick/other_mobs.dm:28]: Keep this in, there's still plenty of stuff that uses COMSIG_HUMAN_EARLY_UNARMED_ATTACK, and we don't want whatever bug this was fixing cropping back up.

**Comment by CoffeeDragon16** [comment]
> Bolas slow them so their only way to do damage is to slowly take off the bola while they are being shot at.

Yeah, lethals. I don't think you should be forced immediately to shooting a guy for minor grief because they have an immunity to stuns.

> The arm breaking of current hulks mean that the constant wall and item breaks is almost a non issue since most dont wanna break their arms just to get into a room

They just break airlocks.

File: code/game/objects/items/devices/scanners.dm [hunk 6]
```diff
@@ -355,7 +353,7 @@ GENE SCANNER
 			|| S.mutantappendix != initial(S.mutantappendix) \
 			|| S.flying_species != initial(S.flying_species)
 
-		render_list += "<span class='info ml-1'>Species: [S.name][mutant ? "-derived mutant" : ""]</span>\n"
+		render_list += "<span class='info m1 -1'>Species: [S.name]</span>\n"
```
Comment (Jared-Fogle) [inline code/game/objects/items/devices/scanners.dm:356]:
```suggestion
		render_list += "<span class='info m1-1'>Species: [S.name]</span>\n"
```

You did it again.

Comment (CoffeeDragon16) [reply code/game/objects/items/devices/scanners.dm:356]: aaaaaa my bad

**Comment by usnpeepoo** [comment]
if anything, if you're inclined about how security has to resort to lethals to deal with hulk just make the bar higher for hulks to get stunned ie more than 1/2 stun baton hits or more disabler shots.

File: code/_onclick/other_mobs.dm (hunk 5 above)
Comment (CoffeeDragon16) [reply code/_onclick/other_mobs.dm:28]: I put that back in. I assumed it was specifically hulk related, but I see why that's important to keep in.

File: code/_onclick/other_mobs.dm [hunk 7]
```diff
@@ -24,6 +24,8 @@
 	var/obj/item/clothing/gloves/G = gloves // not typecast specifically enough in defines
 	if(proximity && istype(G) && G.Touch(A,1))
 		return
+	//This signal is needed to prevent gloves of the north star + hulk.
+	if(SEND_SIGNAL(src, COMSIG_HUMAN_EARLY_UNARMED_ATTACK, A, proximity) & COMPONENT_CANCEL_ATTACK_CHAIN)
```
Comment (Jared-Fogle) [inline code/_onclick/other_mobs.dm:28]: You didn't put the return afterwards.

**Comment by Jared-Fogle** [comment]
Fix the compile errors.

**Comment by wesoda25** [comment]
So make their arms break from airlocks too then 🙄🙄

**Comment by optimumtact** [comment]
no

**Comment by Makarov22** [comment]
> no

AHHAHAHAHA BASED

**Comment by CoffeeDragon16** [comment]
dang!

**Comment by usnpeepoo** [comment]
@optimumtact can u do this for every removal pr pls and thanks

**Comment by Megarop** [comment]
ided

**Pull request closed without merging**